### PR TITLE
feat: mixed conditions

### PR DESCRIPTION
**Pull request by astahmer**
Closes https://github.com/chakra-ui/panda/discussions/1216

## 📝 Description

Add a way to create config conditions with nested at-rules/selectors

```ts
export default defaultConfig({
  conditions: {
    extend: {
      supportHover: ['@media (hover: hover) and (pointer: fine)', '&:hover'],
    },
  },
})
```

```ts
import { css } from '../styled-system/css'

css({
  _supportHover: {
    color: 'red',
  },
})
```

will generate the following CSS:

```css
@media (hover: hover) and (pointer: fine) {
  &:hover {
    color: red;
  }
}
```

## 💣 Is this a breaking change (Yes/No):

no

### CODE REVIEW

**Comment by anubra266** [comment]
We probably want to update this:
CleanShot 2024-02-09 at 23 36 12@2x

File: packages/config/src/validation/validate-condition.ts [hunk 1]
```diff
@@ -1,12 +1,31 @@
...
+      }
+    })
+
+    const hasAtLeastOneAmperand = condition.some((c) => c.includes('&'))
+    if (!hasAtLeastOneAmperand) {
+      addError(
+        'conditions',
+        `Mixed selectors should contain the \`&\` character in at least one segment: \`${condition.join(', ')}\``,
+      )
     }
```
Comment (segunadebayo) [inline packages/config/src/validation/validate-condition.ts:29]: A mixed condition could definitely use two "at-rule" (say @media and @supports). Maybe this is perhaps too strict?

File: packages/core/src/sort-style-rules.ts [hunk 2]
```diff
@@ -1,43 +1,83 @@
...
-  if (a.conditions!.length === b.conditions!.length) {
-    const lastA = a.conditions![a.conditions!.length - 1]
-    const lastB = b.conditions![b.conditions!.length - 1]
 
-    const atRule1 = lastA.params ?? lastA.rawValue
-    const atRule2 = lastB.params ?? lastB.rawValue
+const flatten = (conds: ConditionDetails[]) => conds.flatMap((cond) => (cond.type === 'mixed' ? cond.value : cond))
 
-    if (!atRule1 || !atRule2) return 0
+const compareAtRuleOrMixed = (a: WithConditions, b: WithConditions) => {
```
Comment (segunadebayo) [inline packages/core/src/sort-style-rules.ts:27]: Can you include some tests that cover this new use case? It's a little hard to follow, and I prefer to read the tests instead.

If we could even isolate this bit of the code so we can test it independently, that would be even better.

File: sandbox/preact-ts/panda.config.ts [hunk 3]
```diff
@@ -1,6 +1,11 @@
 import { defineConfig } from '@pandacss/dev'
 
 export default defineConfig({
+  conditions: {
+    mq: '@media (min-width: 640px)',
+    custom: '&[data-custom]',
+    supportHover: ['@media (hover: hover) and (pointer: fine)', '@supports (display: grid)', '&:hover'],
+  },
```
Comment (segunadebayo) [inline sandbox/preact-ts/panda.config.ts:8]: Let's move these to a dedicated test somewhere in packages/core so we can be more confident it works in the future.

File: packages/core/src/sort-style-rules.ts (hunk 2 above)
Comment (astahmer) [reply packages/core/src/sort-style-rules.ts]: I had done that [here](https://github.com/chakra-ui/panda/blob/ae20e36145e4fddea5a9dd53627389ee93e48743/packages/core/__tests__/rule-processor.test.ts#L1296)

I just added a new one specifically for `compareAtRuleOrMixed` [here](https://github.com/chakra-ui/panda/blob/a0a3c1ee028d85b8ed9390f98cf3111a1b1d153d/packages/core/__tests__/conditions.test.ts#L93)

Comment (astahmer) [reply packages/core/src/sort-style-rules.ts]: added further improvements to the sorting logic & added a bunch more tests in [this commit](https://github.com/chakra-ui/panda/pull/2177/commits/7d821dbd5f2bc2ce03f8448471e25fc745954bd1) in the rule-processor.test file
